include/quicr/common.h

```diff
@@ -77,12 +77,13 @@ namespace quicr {
      */
     struct FetchAttributes
     {
-        uint8_t priority;                 ///< Fetch priority
-        messages::GroupOrder group_order; ///< Fetch group order
-        uint64_t start_group;             ///< Fetch starting group in range
-        uint64_t start_object;            ///< Fetch starting object in group
-        uint64_t end_group;               ///< Fetch final group in range
-        uint64_t end_object;              ///< Fetch final object in group
+        uint8_t priority;                 ///< Priority
+        messages::GroupOrder group_order; ///< Group order
+        uint64_t start_group;             ///< Starting group to fetch from, inclusive.
+        uint64_t start_object;            ///< Starting object in start_group to fetch from, inclusive.
+        uint64_t end_group;               ///< Final group to fetch from, inclusive.
+        std::optional<std::uint64_t>
+          end_object; ///< Final object in end_group to fetch from, inclusive. If null, fetch all objects in end_group.
```

Comment:
I'll just state it again here, not sure it needs optional
```suggestion
        std::uint64_t end_object = -1; ///< Final object in end_group to fetch from, inclusive. If invalid, fetch all objects in end_group.
```

Also, not sure I agree with the comment. If this is null? I thought the spec said if group was 0

Reply:
I agree it's not great. If this was something like Swift, something like this might get the intent across better than `?` or a `-1`.

```swift
enum EndFilter { case entireGroup, case endObject(messages::ObjectId) }
EndFilter end_filter;
```

Just for posterity, this value is specifically different to the value that is in the wire message, this is after resolving that back to the inclusive range that the spec talks about. A wire message with an end object of 1 means you want your fetch range (inclusive) to end with and include the 0th object. In that case, end object here *would* be 0, reflecting the inclusive range.

I was trying to make a distinction here between the wire format optimisation of the 0 == whole group vs. endObject -= 1 and the inclusive range the spec refers to.